src/state/slices/color_ramp_slice.ts

```diff
@@ -50,6 +50,8 @@ export type ColorRampSliceState = {
    * `KNOWN_CATEGORICAL_PALETTES`. `null` if the palette has no match.
    */
   categoricalPaletteKey: string | null;
+  /** The current `categoricalPalette`, as a `ColorRamp` object. */
+  categoricalPaletteRamp: ColorRamp;
```

Comment:
This is about keeping an edited color ramp when the user switches away from the categorical feature they were editing on to a continuous feature and back again, right? Maaaybe could use a short comment to clarify that, up to you

Reply:
Mm, it's more to move the instantiation/disposal of the `ColorRamp` so subscribers to the state don't have to worry about doing it themselves. I can clarify the comment, though.